src/resource/tiles/tiles.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, NotFoundException, HttpStatus } from "@nestjs/common";
+import { Model } from "mongoose";
+import { InjectModel } from "@nestjs/mongoose";
+import { Tile } from "./schemas/tile.schema";
+import { CreateTileDto } from "./dto/create-tile.dto";
+import { UpdateTileDto } from "./dto/update-tile.dto";
+
+@Injectable()
+export class TilesService {
+  constructor(@InjectModel(Tile.name) private readonly tileModel: Model<Tile>) {}
+  async findAll(): Promise<Tile[]> {
+    return await (await this.tileModel.find().exec()).filter((tile) => tile.isDeleted !== true);
+  }
+
+  async findOne(id: string): Promise<Tile> {
+    try {
+      const tile = await this.tileModel.findOne({ _id: id }).exec();
+      return tile;
+    } catch {
+      throw new NotFoundException({
+        status: HttpStatus.NOT_FOUND,
+        message: "Tile cannot be find, please search again",
+      });
```

Comment:
double check the global error handling stuff. for NotFoundException, I don't think we should write HTTP status explicitly here.